agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
+    if (!(inst instanceof InvokeInstruction)) {
+      debug_transform.log(
+          "getReplacementInstruction: EXIT %s.%s%n", mg.getClassName(), mg.getName());
+      return null;
+    }
 
-    switch (inst.getOpcode()) {
-      case Const.INVOKESTATIC:
-        {
-          InstructionList il = new InstructionList();
-          INVOKESTATIC is = (INVOKESTATIC) inst;
-          String cname = is.getClassName(pgen);
-          String mname = is.getMethodName(pgen);
-          Type[] args = is.getArgumentTypes(pgen);
-          MethodDef orig = new MethodDef(cname + "." + mname, args);
-          ReplacementClass call = current_method_map.get(orig);
-          if (call != null) {
-            call.cnt++;
-            String classname = call.method_class;
-            String methodname = mname;
-            debug_map.log(
-                "%s.%s: Replacing method %s.%s (%s) with %s.%s%n",
-                mg.getClassName(),
-                mg.getName(),
-                cname,
-                mname,
-                UtilMDE.join(args, ", "),
-                classname,
-                methodname);
-            il.append(
-                ifact.createInvoke(
-                    classname, methodname, is.getReturnType(pgen), args, Const.INVOKESTATIC));
-          }
-          return (il);
-        }
+    InvokeInstruction invocation = (InvokeInstruction) inst;
+    MethodDef orig;
+    try {
+      orig = MethodDef.of(invocation, pgen);
+    } catch (Throwable e) {
 
-      case Const.INVOKEVIRTUAL:
-        {
-          InstructionList il = new InstructionList();
-          INVOKEVIRTUAL iv = (INVOKEVIRTUAL) inst;
-          String cname = iv.getClassName(pgen);
-          String mname = iv.getMethodName(pgen);
-          Type[] args = iv.getArgumentTypes(pgen);
-          Type instance_type = iv.getReferenceType(pgen);
-          Type[] new_args = BCELUtil.insert_type(instance_type, args);
-          MethodDef orig = new MethodDef(cname + "." + mname, args);
-          if (debug_class)
-            System.out.printf("looking for %s in map %s%n", orig, current_method_map);
-          ReplacementClass call = current_method_map.get(orig);
-          if (call != null) {
-            call.cnt++;
-            String classname = call.method_class;
-            String methodname = mname;
-            debug_map.log(
-                "Replacing method %s.%s (%s) with %s.%s%n",
-                cname, mname, ArraysMDE.toString(args), classname, methodname);
-            il.append(
-                ifact.createInvoke(
-                    classname, methodname, iv.getReturnType(pgen), new_args, Const.INVOKESTATIC));
-          }
-          return (il);
-        }
+      debug_transform.log(
+          "getReplacementInstruction: EXIT exception getting method for instruction %s of %s.%s: %s%n",
+          invocation.toString(), mg.getClassName(), mg.getName(), e);
+      return null;
```

Comment:
This seems really serious.  Why not crash rather than proceeding?
If this isn't serious, then explain in the log what could have caused this and what a user can to do prevent it.